src/Microsoft.DotNet.ServiceFabric.ServiceHost/Actors/DelegatedActor.cs

```diff
@@ -295,7 +296,24 @@ protected override IEnumerable<ServiceReplicaListener> CreateServiceReplicaListe
 
         protected override async Task RunAsync(CancellationToken cancellationToken)
         {
-            await base.RunAsync(cancellationToken);
+            var logger = Container.GetRequiredService<ILogger<DelegatedActor>>();
+            try
+            {
+                await using var _ =
+                    cancellationToken.Register(() => logger.LogInformation("Service abort cancellation requested"));
+                logger.LogInformation("Entering service 'RunAsync'");
+                await base.RunAsync(cancellationToken);
+                logger.LogWarning("Abnormal service exit without cancellation");
```

Comment:
This is technically not "abnormal". Services that don't do anything in RunAsync will hit this.

The website is a big one that would hit this I think.

Reply:
Doesn't exiting mean SF will restart? Or are actors special?

Reply:
From these docs, https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-reliable-services-lifecycle#notes-on-the-service-lifecycle

> It is valid for a service to complete RunAsync() successfully and return from it. Completing is not a failure condition. Completing RunAsync() indicates that the background work of the service has finished.